### DISCUSSION TITLE
Strict mode for reading/writing ical

### DISCUSSION THREAD

**Discussion by hukka** [collective/icalendar#548] [Ideas]
Currently icalendar happily reads and writes data without all the fields required by the RFC. There should be a strict mode available, that does error checking.

**Comment by regebro** [comment]
I think a separate checker would be a better idea. It could be a part of icalendar, or separate.

**Comment by hukka** [comment]
That might work for reading, but for writing icalendar needs to be able to raise Exceptions when all required fields are not given.

**Comment by regebro** [comment]
What is the reason you can't just call a verification method before writing?

**Comment by hukka** [comment]
I wasn't even aware of a verification method. I tried looking for it in docs and source now, but still can't find it. Where is it?

**Comment by regebro** [comment]
There isn't one. Which is why I said "a" and not "the" verification method.
I take it that your answer to my question is that you could call one, and then I maintain my position that a checker would be a better idea than a strict mode.

**Comment by hukka** [comment]
Yes, anything that could make it certain that the files are valid according to spec (and thus at least hope for some interoperability).

**Comment by garbas** [comment]
@tpievila i would agree with @regebro to have checker/verification/validation method, just because it leaves implementing "strict" mode to users of library. but this is definetly something we should consider in next release (3.1)

**Comment by garbas** [comment]
oh and would be nice if somebody could champion this, since i'm not sure i'll have the time and enough free time in next months to be motivated to implement this (i just don't have a use-case for this so motivation for it is low).

**Comment by jacobSingh** [comment]
Is there a place where strict mode is defined?  My exported cal isn't google compat and I'm not sure why.

**Comment by geier** [comment]
yes, [RFC 5545](https://tools.ietf.org/html/rfc5545)

**Comment by niccokunzmann** [comment]
This is also related to https://github.com/collective/icalendar/discussions/536
